state_db.js

```diff
@@ -303,11 +304,37 @@ class StateDB {
     }
 
     async verify(root, queries, proof) {
-        return new Promise((resolve, reject) => {
-            state_db_verify.call(this._db, root, queries, proof, (err, result) => {
-                if (err) {
-                    return reject(err);
+        return new Promise((resolve) => {
+            state_db_verify.call(this._db, root, queries, proof, (_, result) => {
+                resolve(result);
+            });
+        });
+    }
+
+
+    async verifyInclusionProof(root, queries, proof) {
+        return new Promise((resolve) => {
+            for (let i = 0; i < queries.length; i++) {
+                if (!isInclusion(queries[i], proof.queries[i])) {
+                    resolve(false);
                 }
+            }
+
+            state_db_verify.call(this._db, root, queries, proof, (_, result) => {
```

Comment:
Another option would be to call `verify` here to avoid code duplication, but maybe this is so small that it's fine.

Reply:
Done.